### ISSUE TITLE
[Bug] Add song to playlist : doesn't see NextcloudMusic Playlist

### ISSUE THREAD

**Issue by potagerGit** [icefields/Power-Ampache-2#162]
I use NextcloudMusic on my computer to create & manage my playlist.  On powerampache2 on my phone, I see well the playlist created on my computer, and can listen to them.  But if I listen to a song (outside of the playlist) and I want to add (from my phone, using the "+" button of the player) that song to a  playlist (previously created on computer) it fails.  It only propose to create a new playlist or add to queue, it doesn't propose the existing playlist.

Am i missing something ?

thanks for your help, your app is great !!!  I unsubscribed from spotify last week thanks to you.

nextcloudmusic server 2.0.1
powerampache2 app v 1.00-68-fdroid - DB:82

**Comment by icefields** [comment]
hello @potagerGit , 
the feature has been there since the first alpha release. Make sure that the user that is adding the songs to the playlist is the same that created the playlist. 
For example if you create the playlists as admin on your computer and make them public, all the other users can see the playlist but cannot edit it.
Let me know if you still have issues with this.
See Video:

https://github.com/user-attachments/assets/a5b4293d-4e9c-4e54-b9ef-ce8d8f68e9de

**Comment by potagerGit** [comment]
https://github.com/user-attachments/assets/7bd7c9e5-825a-41f9-ba51-15f833cdee17

hello, here is a video of my issue.  Ine the "playlist" tab, I see OK all my playslits, but when I play a song, if I hit the "+" button, it only ask "create new" (playlist), but doesn't list all playlitst to choose.  
Where could I look for logs to investigate further ? 

thanks again for your time and your great app !

**Comment by potagerGit** [comment]
and the catalog of playlist is consistent with what I see in Nextcloud Music : 
![image](https://github.com/user-attachments/assets/523d48ac-d3f4-43d9-9bc1-ceadb8598bac)

**Comment by icefields** [comment]
hello, I'm not 100% about how Nextcloud handles playlists (the main focus is with the original Ampache backend), Power Ampache 2 only pulls the playlists that you own and have permission to edit when you press the + button. 
Anyway, I will investigate the issue and I will update this thread. 
ps. I reopened this issue.

**Comment by potagerGit** [comment]
Thank you

**Comment by potagerGit** [comment]
Update
https://mastodon.social/@FLOZz/113621843607537436

It works for him not for me, so must be something special on my install ?

What i changed from default nextcloud music :
Music folder (/music)
Share all this folder with 3 other users.

I cannot think of anything else...

**Comment by icefields** [comment]
Thanks for the update.
I also changed the default folder and it seems to work fine on my installation when I tested it, but I haven't had time to test edge cases yet. 
As mentioned before, Power Ampache 2, when you press the + button to add to playlist, only pulls the playlists that YOU are the owner of, and your user has permission to edit. I'm not sure 100% how the Music plugin handles playlists, maybe it's worth involving the developer of Nextcloud Music.
Anyway, I will investigate the issue and let you know my findings in this thread.

One test you can do is to create a playlist directly from Power Ampache 2, and then see if it appears in the list afterwards.

**Comment by potagerGit** [comment]
https://vid.northbound.online/w/wK3payZMmVKDMep7U31oV9

Hello, in this video you will see the behavior with newly created playlist in the app.  I can only add one song, upon playlist creation.
Same public or private list.

Maybe I could send logs or run a more verbose version to help ?

**Comment by potagerGit** [comment]
I checked the user of my playlist in Nextcloud DB, it is "ColasH" (with uppercase C and H).
I noticed in the app that the user is "colash" (only lowercase).

I logued out & on again in the app, to be sure of typing the correct uppercase, I typed correctly but it stills shows "colash" in the app.  
Could it be a reason for my issue ? 

![image](https://github.com/user-attachments/assets/01969046-038e-46fc-95d1-e4d369b09170)

![image](https://github.com/user-attachments/assets/748d359f-ab18-4a0f-b706-349f25eda50d)

**Comment by potagerGit** [comment]
![image](https://github.com/user-attachments/assets/185cb9ff-68f9-4d5d-a209-d8d43aafb3de)

in addition, if this could help, here are the ampache tables from Nextcloud DB.
Strange to see that I have to user id in NC pointing to the same ampache_user_id

**Comment by icefields** [comment]
@potagerGit that is definitely the issue. The username should not be case sensitive. 
On my side, I do have a low priority ticket to handle mixed case usernames, but we already had a conversation about this with the developer of Nextcloud Music and the developers for the Ampache backend, and the backend should work regardless.
To make it short, this is a fix for Nextcloud Music, Ampache compatible backends should handle username as non case-sensitive strings. I will contact the developer later today to ask to make the fix on Music.
Thanks for debugging and finding the issue!

**Comment by icefields** [comment]
actually, we can add him to this thread. @paulijar can you take a look please? looks like the case sensitive usernames issue is still affecting playlist creation and filtering.
Thanks

**Comment by paulijar** [comment]
So what, exactly, do you think that Nextcloud Music app does here which it shouldn't do?

When PA2 "only pulls the playlists that you own and have permission to edit", then how does it do that? Is it possible that you compare the property `owner` of the playlist to the lower-cased username of the logged-in user, since you internally store the username that way?

**Comment by icefields** [comment]
@paulijar I'm currently out, so I'm going with what I remember. If I remember correctly, there's a call to fetch playlists owned by a certain user, something like `GET action=playlists&owner=username` , getting playlists for `joHn` or `john` or `John` should result in the same output.
This said, I checked the official Ampache documentation and it doesn't seem updated with the new parameter to limit the fetching to a single user, so chances are that the call with the new parameter `owner` is not implemented by Music, this means PA2 is falling back to the method that fetches all the playlists and then does the filtering to get the owned lists locally. If that's the case, the bug is on my side and I will fix it in the next release (and I probably pulled you in this thread prematurely) .
I will verify and eventually fix later today.

**Comment by paulijar** [comment]
Yeah, if there's an argument like `owner=username` then NC Music will simply disregard it. Shared playlists are not "a thing" on NC Music (*) i.e. all the playlists available to the user are owned by that same user.

(*) At least yet, there have been some related feature requests. But I don't see this happening in the near future, at least.

**Comment by potagerGit** [comment]
Thanks to both of you for helping make your great apps even better !

**Comment by icefields** [comment]
@potagerGit thank you for helping debugging.
Would you be able to test the fix? 
This is the Github release that I will post officially between tomorrow and Wednesday.
https://github.com/icefields/Power-Ampache-2/raw/refs/heads/version_1.00-71/app/Github/release/app-Github-release.apk

Let me know, if you can. Thanks.

**Comment by potagerGit** [comment]
Yes i tested it, and it works ! 
Thanks again.
